### DISCUSSION TITLE
How to backup Volumes using command line

### DISCUSSION THREAD

**Discussion by enricosoft** [containers/podman#23054] [General]
I'm trying to backup all the volumes related to a container.
The container is running as part of docker-compose.
I'm running podman on windows server 2022.

I found the podman doc about how to export volume to external tar file.
[https://docs.podman.io/en/latest/markdown/podman-volume-export.1.html](https://docs.podman.io/en/latest/markdown/podman-volume-export.1.html)

I tried these and I attached the related error for each line:

podman volume export myVolName --output uploads.tar **// Error: accepts 1 arg(s), received 3**
podman volume export myVolName -o=uploads.tar **// Error: accepts 1 arg(s), received 2**
podman volume export myVolName > uploads.tar **// Error: cannot use command "podman volume export" with the remote podman client**

Seems that the documentation is not updated for example 1 and 2.

For sample 3 I don't understand why I can't run that command from the podman CLI installed on the same server.
Should I run the container with privileges? If so, how?
Does Podman Desktop have its own terminal to run this kind of commands? From GUI I cannot performs actions on volumes... I cannot select a volume and I don't see any menu in "actions" column

Thanks in advance

**Comment by marcokrikke** [comment]
I managed to export/import volumes like this with a remote Podman on MacOS. An example below for exporting volumes from Podman 4 and importing them again into Podman 5.

### Exporting

- SSH into the Podman machine: `podman machine ssh`
- Export the volume(s): `core@localhost:~$ podman volume export <volumename> -o export.tar`
- Exit the session (Ctrl+D)
- Get info on the system connections: `podman system connection list`

````
➜  ~ podman system connection list                                                        
Name                         URI                                                         Identity                                        Default
podman-machine-default       ssh://core@127.0.0.1:64418/run/user/501/podman/podman.sock  /Users/user/.ssh/podman-machine-default  true
podman-machine-default-root  ssh://root@127.0.0.1:64418/run/podman/podman.sock           /Users/user/.ssh/podman-machine-default  false
````

- SCP the files from the machine to your host:`scp -i ~/.ssh/podman-machine-default -P 64418 core@localhost:~/export.tar .`

### Importing

- (In my case: upgrade Podman first)
- Create the volume, for example on the command line or via the GUI
- Get info on the system connections again (they have changed between Podman 4 and 5)

````
➜  ~ podman system connection list                                                                                             
Name                         URI                                                         Identity                                                           Default     ReadWrite
podman-machine-default       ssh://core@127.0.0.1:55674/run/user/501/podman/podman.sock  /Users/user/.local/share/containers/podman/machine/machine  false       true
podman-machine-default-root  ssh://root@127.0.0.1:55674/run/podman/podman.sock           /Users/user/.local/share/containers/podman/machine/machine  true        true
````

- SCP the files from your host to the machine:`scp -i /Users/user/.local/share/containers/podman/machine/machine -P 55674 export.tar root@localhost:~`
- SSH into the Podman machine again: `podman machine ssh`
- Import the volume(s): `root@localhost:~$ podman volume import export.tar <volumename>`

**Comment by nalind** [comment]
~`podman volume export` expects the name of the volume to be last, after options like `--output`.  The order does matter to it.~
